builtin/providers/google/resource_sql_database_instance.go

```diff
@@ -579,6 +603,24 @@ func resourceSqlDatabaseInstanceRead(d *schema.ResourceData, meta interface{}) e
 		_settings["crash_safe_replication"] = settings.CrashSafeReplicationEnabled
 	}
 
+	if v, ok := _settings["disk_autoresize"]; ok && v != nil {
+		if v.(bool) {
+			_settings["disk_autoresize"] = settings.StorageAutoResize
+		}
+	}
+
+	if v, ok := _settings["disk_size"]; ok && v != nil {
+		if v.(int) > 0 && settings.DataDiskSizeGb < int64(v.(int)) {
```

Comment:
Wait, I'm confused. Can you tell me what's wrong with the way I'm reading this?

Say we start with a size of 20GB.

If auto resize increases it to 30, then we do an update for some other reason:
* the read would notice that the size increased, and update the state file
* the next update after that wouldn't do anything because the sizes would match

If auto resize decreases it to 10, then we do an update:
* the read wouldn't update the state file because what was read from the api was less than what's currently there
* the next update after that would try to reduce the size because the smaller one is what's in the state file